src/app/web-app.component.ts

```diff
@@ -45,15 +49,24 @@ export class WebAppComponent implements OnInit {
    * @param {ThemeStorageService} themeStorageService Theme Storage Service.
    * @param {MatSnackBar} snackBar Material Snackbar for notifications.
    * @param {AlertService} alertService Alert Service.
+   * @param {KeyboardShortcutsService} keyboardShortcuts KeyboardS Shortcuts Service.
    */
+
+  private keyboardShortcuts: KeyboardShortcutsService;
+  private unlisten: Unlisten;
```

Comment:
What is the need of this variable?

Reply:
`unlisten` is an instance of the empty interface to which the shortcut bindings are added as an array and this object gets destroyed after the app is closed and thus stops listening to the shorcuts.

Reply:
Won't the object get destroyed automatically anyway once the app is closed?

Reply:
Currently, you are not destroying it manually. The variable is just being assigned initially.

Reply:
Sorry i forgot to answer this one, i was mistaken, now i've updated the comment, not destroying it 
but it is removing the listener from the listeners' list, which can be seen from this part of the code:
```
const unlisten = (): void => {
   this.removeListener(listener);
};
```
it is used to remove the listener from the listener's list which are being handled in the current zone of the app.

Reply:
Sorry i forgot to answer this one, i was mistaken, now i've updated the comment, not destroying it but it is removing the listener from the listeners' list, which can be seen from this part of the code:
```
const unlisten = (): void => {
   this.removeListener(listener);
};
```
it is used to remove the listener from the listener's list which are being handled in the current zone of the app.

Reply:
I am unable to locate this part of code. Where is it? Also, mention a comment here as well.

Reply:
This part is in `keyboard-shortcuts.service.ts`